fix: MOTD `CenterWindow` function in `blakserv` win interface

adrienlaws opened the pull request:
# What

- makes MOTD pop-up dialog box appear centered on the `blakserv` application window on Windows
- modifies `CenterWindow` function in`blakserv` Windows application
- adds support for multiple monitors

# Why

- #608 fixed issues with the Meridian 59 client, this adds similar fixes to the Meridian 59 `blakserv` application
- Fixes the MOTD dialog box appearing on the wrong screen 
    - for example, screen 1 has the application but the MOTD dialog box will open on screen 2

# Notes

- The "Are you sure you want to exit?" dialog box will appear on the correct monitor, but centered on the desktop not the application
  - I looked into fixing this, but that message box is a system-managed dialog box which is difficult to modify
    - I tried a few workarounds but since it does open on the correct monitor, it seems ok
- I also looked into trying to have the application "remember" where the application was placed on the desktop, so it would re-open in the same spot
  - I looked into how to save the position in the registry but it also seemed like I was adding too much complexity since the server should really be run from Linux in production

# Examples

### Before
- MOTD appears on first monitor, always, even when the application is on the second monitor
https://github.com/user-attachments/assets/a0885076-6a85-4151-8703-9dad9a25b5c1


### After
- MOTD opens centered on the application
https://github.com/user-attachments/assets/c96e9f79-8d92-40a0-97ec-df24d0aa290e

akirmse commented on blakserv/interface.c:
```diff
@@ -1027,50 +1028,78 @@ void InterfaceReloadSystem()
+    RECT rcDlg, rcParent, rcScreen;
+    int x, y;
+
+    #define min(a, b) ((a) < (b) ? (a) : (b))
```
Ugh... use std::min and std::max instead of macros.

akirmse commented on blakserv/interface.c:
```diff
@@ -1027,50 +1028,78 @@ void InterfaceReloadSystem()
+    SetWindowPos(hwnd, NULL, x, y, 0, 0, SWP_NOSIZE | SWP_NOACTIVATE);
+
+    #undef min
+    #undef max
```
Remove these.

adrienlaws replied:
Removed

akirmse commented on blakserv/interface.c:
```diff
@@ -1027,50 +1028,78 @@ void InterfaceReloadSystem()
 	{
 	case WM_INITDIALOG :
-		CenterWindow(hwnd,NULL);
+        CenterWindow(hwnd, hwndParent);
```
simplify to 
```
CenterWindow(hwnd, GetParent(hwnd));
```

since there's no other use for the parent window.

adrienlaws replied:
Simplified as requested.

akirmse commented on blakserv/interface.c:
```diff
@@ -586,6 +586,7 @@ void StartupComplete()
 void InterfaceCreate(HWND hwnd,WPARAM wParam,LPARAM lParam)
 {
 	SetWindowText(hwnd,BlakServNameString());
+	CenterWindow(hwnd, NULL);
```
I'm not sure this is a good idea.  Does this always center the server window on the monitor?  I certainly don't want that; I want it to come up where it currently does.

adrienlaws replied:
Removing as requested.

akirmse commented:
Nice fix, though I don't want the app centered on my monitor.  (For one thing, it will be hidden by the client if I do that, which will make local testing harder).  I believe there are other shell-related ways to control initial location.

adrienlaws commented:
This is ready for another review, when possible.

I have addressed all the comments and removed the automatic centering of the application on the desktop.  

The `blakserv` application starts in it's original position now, but the MOTD dialog box will appear centered on the `blakserv` application.

akirmse commented on blakserv/interface.c:
```diff
@@ -22,6 +22,7 @@
 	
 */
 
+#define NOMINMAX
```
Are you sure this is needed?  It looks like the original code used std::max and didn't have this.

Precompiled headers in C++ is a complicated topic, but extra #defines like this can slow down compilation.

adrienlaws replied:
Removed, you are right.
